CSS code hints in 'style' attribute value context

swmitra opened the pull request:
This PR enables Brackets to provide CSS code hints property-value code hints while editing 'style' attribute value in 'html' mode. The implementation contains minor tweaks to the CSSUtils module to recognize this as 'css' context as CodeMirror doesn't understand the 'style' attribute value as 'css' context. 

Adding @ingorichter @petetnt @zaggino and @ficristo for review, but others can also join 👍

swmitra committed:
CSS code hints in 'style' attribute value context

ficristo commented:
Sorry but I don't have enough interest in this feature to review it.
Maybe @humphd is interested?

swmitra commented:
@humphd Can you please have a look at this PR?

ingorichter commented on src/language/CSSUtils.js:
```diff
@@ -613,16 +620,34 @@ define(function (require, exports, module) {
+        // and in attribute value state of a tag with attribute name style
+        if (ctx.token.state.htmlState && (!ctx.token.state.localMode || ctx.token.state.localMode.name !== "css")) {
+
+                // tagInfo is required to aquire the style attr value
```
nit: indent level

petetnt replied:
Indent

ingorichter commented:
@swmitra it would be beneficial to have some unit tests for the new functionality. This is complex and might break accidentally.

swmitra commented:
Sure @ingorichter. I will add some functional tests in this PR and ping you.

petetnt reviewed (changes requested):
Change set LGTM with a small nit. 

However it doesn't seem to handle multiline styles yet:

```html
<div style="background-color: red; // works here
color: white; // doesn't work here>Foo</div>
```

It would be also great if it worked on JSX context (`<div style={{backgroundColor: red}}`>Hey</div>`) but that's more a feature request :)

swmitra committed:
Handle multiline attribute value and add UNIT test for element styles

swmitra commented:
@ingorichter I have added some unit tests in my last commit.
@petetnt Fixed the multiline attribute value issue and corrected indentation. 
Can you guys please review it once more? ☺️

zaggino reviewed (approved):
- checked the code, seems good
- checked the unit tests, present & passing
- tested the feature manually, working fine

ingorichter commented:
I'm too late :-/
